src/realm/object-store/sync/async_open_task.cpp

```diff
@@ -32,6 +33,46 @@ AsyncOpenTask::AsyncOpenTask(std::shared_ptr<_impl::RealmCoordinator> coordinato
 {
 }
 
+void AsyncOpenTask::handle_realm_async_open(Callback&& callback, std::shared_ptr<_impl::RealmCoordinator> coordinator,
+                                            Status status)
+{
+    if (!status.is_ok())
+        return callback({}, std::make_exception_ptr(Exception(status)));
+
+    ThreadSafeReference realm;
+    try {
+        realm = coordinator->get_unbound_realm();
+    }
+    catch (...) {
+        return callback({}, std::current_exception());
+    }
+
+    auto config = coordinator->get_config();
+    if (config.sync_config && config.sync_config->flx_sync_requested) {
+        using namespace sync;
+        const auto subscription_initializer = config.sync_config->subscription_initializer;
+        const auto always_run = config.sync_config->always_run;
+        if (subscription_initializer && always_run) {
+            auto shared_realm = coordinator->get_realm();
```

Comment:
I am not sure if this is the right thing to do here, Essentially I need to get some realm that is not unbound, but I still need to retain the unbound one for notifying the callback when async open has finished processing. 
Am I doing the right thing? @tgoyne @danieltabacaru @michael-wb